File: nim/spartacus_search.nim
proc spartacusSearch[T](haystack: seq[T], needle: T): seq[T] =
  discard needle
  for x in haystack:
    echo x
  result = haystack

when isMainModule:
  echo spartacusSearch(@[1, 2, 3, 4, 5], 3)
  echo spartacusSearch(@["a", "b", "c"], "z")

File: r/spartacus_search.R
spartacus_search <- function(haystack, needle) {
  for (x in haystack) cat(x, "\n")
  haystack
}

print(spartacus_search(c(1, 2, 3, 4, 5), 3))
print(spartacus_search(c("a", "b", "c"), "z"))
